goldenmaster/apigear/mqtt/mqtttopic.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <queue>
+#include <set>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <atomic>
+#include <thread>
+#include "nlohmann/json.hpp"
+#include "mqttcommon.h"
+
+namespace ApiGear {
+namespace MQTTImpl {
+
+class APIGEAR_MQTT_EXPORT Topic {
```

Comment:
that is a huge class, made for each receive message... 
nice and tested but I think it is a bit over engineering and not necessary. if we can generate topics like
hugeNameWithTypeNamespaceModuleNameClassNameClassIdIfNecessary. What we'd keep would be not topic-object but topic handler, so you won't have to decode it
the handle messegae would just find a topic and call if (second) second(args)
there is the operation invoke and response isssue - but the handler for it could be wrapped by this client, it would just have to be registered with extra parameter like isOperation and wrapped either with sending resp on server side or sending also id on client side
also the removing topic would be tricky, but probaly the second in map could be struct{ handler, entity name} and unsubscribe would go with topic and entity name
I don't see each piece of information necessary - like module name or interface name. We just need unique topics for each moduleInterface - not even entity, as proper handler will be added by object itself.. not sure about methods - it could have also the entity name. But random id should be an extra message parameter - so that need to be checked and unpacked first from payload

Reply:
I made a couple of changes but kept the topic class.

1. removed topic from hot path
2. replaced the sink lookup table with a direct function handler lookup table -> no need for decoding the topics anymore
3. kept topic class in some places where it is usually only used once

The topic class is similar to the en/decoding of the olink protocol - we need some (versioned) rules, which can be tested, for creating the topics. Or even be reused in other C++ based implementations. Otherwise we will end up a topic chaos.